peering/peering.go

```diff
@@ -242,6 +243,22 @@ func (ps *PeeringService) AddPeer(info peer.AddrInfo) {
 	}
 }
 
+// ListPeer lists peers in the peering service.
+func (ps *PeeringService) ListPeer() []peer.AddrInfo {
+	out := make([]peer.AddrInfo, len(ps.peers))
+	c := 0
+	for k, v := range ps.peers {
+		out_addrs := make([]multiaddr.Multiaddr, len(v.addrs))
+		copy(out_addrs, v.addrs)
+		out[c] = peer.AddrInfo{
+			ID:    k,
+			Addrs: out_addrs,
+		}
+		c++
```

Comment:
It's generally better to just use append, rather than manually indexing. That would also let you simplify this code a bit:


```go
out := make([]peer.AddrInfo, 0, len(ps.peers))
for id, addrs := range ps.peers {
    ai := peer.AddrInfo{ID: id}
    ai.Addrs = append(pi.Addrs, peers...) // Will copy. Probably not strictly necessary in this case.
    out = append(out, ai)
}
```

Reply:
I totally agree. I will correct it.